### PR TITLE
Add INTEL/SPV_INTEL_fpga_buffer_location.asciidoc

### PR DESCRIPTION
**Pull request by danielz1intel**
Add INTEL/SPV_INTEL_fpga_buffer_location.asciidoc

### CODE REVIEW

**Review by bashbaug — APPROVED** [review]
LGTM, thanks!  Needs a non-Intel reviewer to merge.

File: extensions/INTEL/SPV_INTEL_fpga_buffer_location.asciidoc [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+== Dependencies
+
+This extension is written against the SPIR-V Specification,
+Version 1.6 Revision 2.
+
+This extension requires SPIR-V 1.0.
+
+== Overview
+
+This extension adds a function parameter decoration that is useful for FPGA targets.  This decoration indicates that a particular global memory pointer can only access a particular physical memory location.  Knowing this information at compile time can allow FPGA compilers to generate load store units of lower area for accesses done through such a pointer.
```
Comment (MrSidims) [inline extensions/INTEL/SPV_INTEL_fpga_buffer_location.asciidoc:43]: Taking an account https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/1801/files - the overview should be updated

File: extensions/INTEL/SPV_INTEL_fpga_buffer_location.asciidoc [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+=== Decoration
+
+Modify Section 3.20, Decoration, adding these rows to the Decoration table:
+
+--
+[options="header"]
+|====
+2+^| Decoration | Extra Operands | Enabling Capabilities
+| 5921 | *BufferLocationINTEL* + 
+Apply only to a pointer. Indicates that the pointer must only point into the physical memory identified by the subsequent literal number operand. The exact semantics of the literal number are implementation defined. | Literal +
```
Comment (MrSidims) [inline extensions/INTEL/SPV_INTEL_fpga_buffer_location.asciidoc:82]: Shouldn't it be:

```suggestion
Apply only to a pointer. Indicates that the pointer and all of its aliases must only point into the physical memory identified by the subsequent literal number operand. The exact semantics of the literal number are implementation defined. | Literal +
```
?

File: extensions/INTEL/SPV_INTEL_fpga_buffer_location.asciidoc (hunk 1 above)
Comment (broxigarchen) [reply extensions/INTEL/SPV_INTEL_fpga_buffer_location.asciidoc:43]: Right. I think this should be both function parameter decoration and Instruction parameter decoration.

There are some history here. The function parameter decoration is already checked in into the code but the spec is not yet posted at that time... So this spec actually covers both

File: extensions/INTEL/SPV_INTEL_fpga_buffer_location.asciidoc (hunk 2 above)
Comment (broxigarchen) [reply extensions/INTEL/SPV_INTEL_fpga_buffer_location.asciidoc:82]: I think it makes sense but I am not sure if this is necessary. Wouldn't alias the word itself already implicitly indicates that these two pointers are equivalent?

File: extensions/INTEL/SPV_INTEL_fpga_buffer_location.asciidoc (hunk 1 above)
Comment (broxigarchen) [inline extensions/INTEL/SPV_INTEL_fpga_buffer_location.asciidoc:43]:
```suggestion
This extension adds a decoration for a function parameter or a pointer argument of memory access instructions that is useful for FPGA targets.  This decoration indicates that a particular global memory pointer can only access a particular physical memory location.  Knowing this information at compile time can allow FPGA compilers to generate load store units of lower area for accesses done through such a pointer.  
```

Comment (broxigarchen) [reply extensions/INTEL/SPV_INTEL_fpga_buffer_location.asciidoc:43]: maybe something like this?

File: extensions/INTEL/SPV_INTEL_fpga_buffer_location.asciidoc (hunk 2 above)
Comment (MrSidims) [reply extensions/INTEL/SPV_INTEL_fpga_buffer_location.asciidoc:82]: After some thinking - I no longer have objections of leaving this as is.

File: extensions/INTEL/SPV_INTEL_fpga_buffer_location.asciidoc (hunk 1 above)
Comment (danielz1intel) [reply extensions/INTEL/SPV_INTEL_fpga_buffer_location.asciidoc]: Please review the recent commit to see if the displayed is appropriate. Thanks!

Comment (MrSidims) [reply extensions/INTEL/SPV_INTEL_fpga_buffer_location.asciidoc]: Moved the discussion internally. Once internal discussion resolved please don't forget to update generated .html file here.

Comment (danielz1intel) [reply extensions/INTEL/SPV_INTEL_fpga_buffer_location.asciidoc]: Adjusted the docs in this PR after internal adjustments.

**Review by MrSidims — APPROVED** [review]
LGTM

**Review by broxigarchen — APPROVED** [review]

**Pull request merged**